### PR TITLE
Update dataset_formats.mdx

### LINKED ISSUE #2219: GKD trainer missing from docs/dataset formats
GKD trainer is missing here https://huggingface.co/docs/trl/en/dataset_formats#which-dataset-format-to-use

### PR DESCRIPTION
**Pull request by August-murr**
# What does this PR do?

Fixes #2219
The trainers on [TRL docs from the website](https://huggingface.co/docs/trl/en/dataset_formats#which-dataset-format-to-use ) have links attached, but [the markdown file in the repo](https://github.com/huggingface/trl/blob/main/docs/source/dataset_formats.mdx) didn't contain any of the links. So, I wasn't sure If I should add the [`GKDTrainer`](https://huggingface.co/docs/trl/v0.11.3/en/gkd_trainer#trl.GKDTrainer) docs link to the table, please let me know if I need to add it to this PR.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [ ] Did you write any new necessary tests?

## Who can review?

@qgallouedec

### CODE REVIEW

File: docs/source/dataset_formats.mdx [hunk 1]
```diff
@@ -209,6 +209,7 @@ Choosing the right dataset format depends on the task you are working on and the
 | [`RewardTrainer`]       | [Preference (implicit prompt recommended)](#preference) |
 | [`SFTTrainer`]          | [Language modeling](#language-modeling)                 |
 | [`XPOTrainer`]          | [Prompt-only](#prompt-only)                             |
+| [`GKDTrainer`]          | [Conversational](#conversational-dataset-format)        |                    |
```
Comment (qgallouedec) [inline docs/source/dataset_formats.mdx:212]: conversational is a *type*, not a *format*. I think GKD needs prompt only but you'll need to double check

**Comment by qgallouedec** [comment]
> The trainers on [TRL docs from the website](https://huggingface.co/docs/trl/en/dataset_formats#which-dataset-format-to-use) have links attached, but [the markdown file in the repo](https://github.com/huggingface/trl/blob/main/docs/source/dataset_formats.mdx) didn't contain any of the links. So, I wasn't sure If I should add the [`GKDTrainer`](https://huggingface.co/docs/trl/v0.11.3/en/gkd_trainer#trl.GKDTrainer) docs link to the table, please let me know if I need to add it to this PR.


When you write ```[`GKDTrainer`]```, the link is automatically create, no need to add it

File: docs/source/dataset_formats.mdx (hunk 1 above)
Comment (qgallouedec) [reply docs/source/dataset_formats.mdx:212]: please also sort it alphabeticaly

Comment (August-murr) [reply docs/source/dataset_formats.mdx:212]: > I think GKD needs prompt-only but you'll need to double check

>no GKD needs prompts as well as full conversations in the "messages" key,

in [GKDTrainer docs](https://huggingface.co/docs/trl/v0.11.3/en/gkd_trainer#expected-dataset-format) the expected data format is described as:
The dataset should be formatted as a list of “messages” where each message is a list of dictionaries with the following keys:

- role: either system, assistant or user
- content: the message content

like:
```python
messages = [
    {"role": "user", "content": "Hello, how are you?"},
    {"role": "assistant", "content": "I'm doing great. How can I help you today?"},
    {"role": "user", "content": "I'd like to show off how chat templating works!"},
]
```

which is why I wrote it as **Conversational**

is the GKDTrainer docs incorrect?

Comment (qgallouedec) [reply docs/source/dataset_formats.mdx]: > is the GKDTrainer docs incorrect?

it's not incorrect but it has to be updated. Don't worry about that, we will make this clarification in a follow-up PR

File: docs/source/dataset_formats.mdx [hunk 2]
```diff
@@ -208,7 +209,7 @@ Choosing the right dataset format depends on the task you are working on and the
 | [`PPOv2Trainer`]        | Tokenized language modeling                             |
 | [`RewardTrainer`]       | [Preference (implicit prompt recommended)](#preference) |
 | [`SFTTrainer`]          | [Language modeling](#language-modeling)                 |
-| [`XPOTrainer`]          | [Prompt-only](#prompt-only)                             |
+| [`XPOTrainer`]          | [Prompt-only](#prompt-only)                             |
```
Comment (qgallouedec) [inline docs/source/dataset_formats.mdx:212]:
```suggestion
| [`XPOTrainer`]          | [Prompt-only](#prompt-only)                             |
```

**Review by qgallouedec — APPROVED** [review]

**Comment by qgallouedec** [comment]
Lgtm thanks @August-murr!

**Pull request merged**